src/script/lua_api/l_util.cpp

```diff
@@ -423,6 +425,31 @@ int ModApiUtil::l_get_version(lua_State *L)
 	return 1;
 }
 
+int ModApiUtil::l_sha1(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	std::string data = luaL_checkstring(L, 1);
+	bool hex = !lua_isboolean(L, 2) || !lua_toboolean(L, 2);
+
+	// Compute actual checksum of data
+	std::string data_sha1;
+	{
+		SHA1 data_sha1_calculator;
+		data_sha1_calculator.addBytes(data.c_str(), data.size());
+		unsigned char *data_tmpdigest = data_sha1_calculator.getDigest();
+		data_sha1.assign((char*) data_tmpdigest, 20);
+		free(data_tmpdigest);
+	}
+
+	if (hex) {
+		std::string sha1_hex = hex_encode(data_sha1.c_str());
+		lua_pushlstring(L, sha1_hex.data(), sha1_hex.size());
```

Comment:
equivalent to `lua_pushtring(L, sha1_hex.c_str());`

Reply:
True, I guess the hex version isn't going to have any nulls.